components/director/internal/domain/application/service.go

```diff
@@ -1006,6 +1006,26 @@ func (s *service) ensureApplicationNotPartOfScenarioWithRuntime(ctx context.Cont
 	return nil
 }
 
+// ensureApplicationNotPartOfAnyScenario Checks if an application has scenarios associated with it. If the application is
+// associated with any scenario it can not be deleted before unassigning from that scenario
+func (s *service) ensureApplicationNotPartOfAnyScenario(ctx context.Context, tenant, appID string) error {
+	scenarios, err := s.getScenarioNamesForApplication(ctx, appID)
+	if err != nil {
+		return err
+	}
+
+	if len(scenarios) > 0 {
+		application, err := s.appRepo.GetByID(ctx, tenant, appID)
+		if err != nil {
+			return errors.Wrapf(err, "while getting application with id %s", appID)
+		}
+		msg := fmt.Sprintf("System %s is still used and cannot be deleted. Unassign the system from the following formations first: %s.", application.Name, strings.Join(scenarios, ", "))
```

Comment:
Use generic message. We will reuse this method to prevent merge of systems  if there are existing formations associated with the source system